graphql_compiler/compiler/ir_lowering_match/ir_lowering.py

```diff
@@ -313,16 +324,14 @@ def visitor_fn(expression):
             # If the root_block is a Backtrack, translate its Location if necessary.
             if isinstance(new_step.root_block, Backtrack):
                 old_location = new_step.root_block.location
-                if old_location in location_translations:
-                    new_location = location_translations[old_location]
-                    new_step = new_step._replace(root_block=Backtrack(new_location))
+                new_location = location_translations.get(old_location, old_location)
+                new_step = new_step._replace(root_block=Backtrack(new_location))
```

Comment:
Why make this change? If the location of the `Backtrack` doesn't change, we don't need to call `._replace()` at all -- that's just wasteful in memory.